### PR TITLE
Use `setuptools_scm` to set the project version

### PR DESCRIPTION
**Pull request by aalexandrov**
*Issue #, if available:*

Derive the verion used in `gluonts.__version__` automatically from the latest tag along the parent chain.

See https://pypi.org/project/setuptools-scm/ for details.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-136/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-136/1/index.html

**Comment by szha** [comment]
Job PR-136/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-136/2/index.html

File: requirements-setup.txt [hunk 1]
```diff
@@ -1,13 +1,15 @@
 black==19.3b0
 click==6.*
 mypy==0.630
+notedown
```
Comment (jaheba) [inline requirements-setup.txt]: Is this part of this commit?

**Comment by szha** [comment]
Job PR-136/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-136/4/index.html

File: requirements-setup.txt (hunk 1 above)
Comment (aalexandrov) [reply requirements-setup.txt]: No, this this is in a separate commit but I included it here as I didn't want to have a one-liner PR.

**Pull request merged**